src/mlpack/bindings/go/mlpack/capi/io_util.cpp

```diff
@@ -18,186 +18,250 @@ namespace mlpack {
 {
-  return IO::GetParam<double>(identifier);
+  util::Params& p = *((util::Params*) params);
+  return p.Get<double>(identifier);
 }
 
 /**
  * Get the int parameter associated with specified identifier.
  */
-int mlpackGetParamInt(const char* identifier)
+int mlpackGetParamInt(void* params, const char* identifier)
 {
-  return IO::GetParam<int>(identifier);
+  util::Params& p = *((util::Params*) params);
+  return p.Get<int>(identifier);
 }
 
 /**
  * Get the bool parameter associated with specified identifier.
  */
-bool mlpackGetParamBool(const char* identifier)
+bool mlpackGetParamBool(void* params, const char* identifier)
 {
-  return IO::GetParam<bool>(identifier);
+  util::Params& p = *((util::Params*) params);
+  return p.Get<bool>(identifier);
 }
 
 /**
  * Get the vector<int> parameter associated with specified identifier.
  */
-void* mlpackGetVecIntPtr(const char* identifier)
+void* mlpackGetVecIntPtr(void* params, const char* identifier)
 {
-  const size_t size = mlpackVecIntSize(identifier);
+  const size_t size = mlpackVecIntSize(params, identifier);
   long long* ints = new long long[size];
 
+  util::Params& p = *((util::Params*) params);
   for (size_t i = 0; i < size; i++)
-    ints[i] = IO::GetParam<std::vector<int>>(identifier)[i];
+    ints[i] = p.Get<std::vector<int>>(identifier)[i];
 
   return ints;
 }
 
 /**
  * Get the vector<string> parameter associated with specified identifier.
  */
-const char* mlpackGetVecStringPtr(const char* identifier, const size_t i)
+const char* mlpackGetVecStringPtr(void* params,
+                                  const char* identifier,
+                                  const size_t i)
 {
-  return IO::GetParam<std::vector<std::string>>(identifier)[i].c_str();
+  util::Params& p = *((util::Params*) params);
+  return p.Get<std::vector<std::string>>(identifier)[i].c_str();
 }
 
 /**
  * Get the vector<int> parameter's size.
  */
-int mlpackVecIntSize(const char* identifier)
+int mlpackVecIntSize(void* params, const char* identifier)
 {
-  return IO::GetParam<std::vector<int>>(identifier).size();
+  util::Params& p = *((util::Params*) params);
+  return p.Get<std::vector<int>>(identifier).size();
 }
 
 /**
  * Get the vector<string> parameter's size.
  */
-int mlpackVecStringSize(const char* identifier)
+int mlpackVecStringSize(void* params, const char* identifier)
 {
-  return IO::GetParam<std::vector<std::string>>(identifier).size();
+  util::Params& p = *((util::Params*) params);
+  return p.Get<std::vector<std::string>>(identifier).size();
 }
 
 /**
  * Set parameter as passed.
  */
-void mlpackSetPassed(const char* name)
+void mlpackSetPassed(void* params, const char* name)
 {
-  IO::SetPassed(name);
+  util::Params& p = *((util::Params*) params);
+  p.SetPassed(name);
 }
 
 /**
  * Reset the status of all timers.
  */
 void mlpackResetTimers()
 {
-  IO::GetSingleton().timer.Reset();
+//  IO::GetSingleton().timer.Reset();
```

Comment:
Not needed anymore?

Reply:
Nice point---I think I can remove all of the code related to this function.  I tried in 0bc423c1